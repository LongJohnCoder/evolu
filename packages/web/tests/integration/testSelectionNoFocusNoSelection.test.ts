import { pageDom, pageGoto, pageKeyboard } from './helpers';

beforeEach(async () => {
  await pageGoto('testSelectionNoFocusNoSelection');
});

test('tab then key arrow right', async () => {
  await pageKeyboard.press('Tab');
  await expect(await pageDom()).toMatchSnapshot();

  await pageKeyboard.press('ArrowRight');
  await expect(await pageDom()).toMatchSnapshot();
});
