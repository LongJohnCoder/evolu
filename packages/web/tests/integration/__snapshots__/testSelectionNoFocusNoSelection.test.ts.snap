// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tab then key arrow right 1`] = `
<div
  id="__next"
>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size:24px"
      >
        ·heading·
      </div>
      <div
        style="font-size:16px"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-1782790912"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-1782790912"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true,"selection":null}·
    </span>
    <span
      class="jsx-1782790912"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false,"selection":null}·
    </span>
  </pre>
</div>
`;

exports[`tab then key arrow right 2`] = `
<div
  id="__next"
>
  <div
    autocorrect="off"
    contenteditable="true"
    data-gramm="true"
    role="textbox"
    spellcheck="false"
    tabindex="0"
  >
    <div
      class="root"
    >
      <div
        style="font-size:24px"
      >
        ·heading·
      </div>
      <div
        style="font-size:16px"
      >
        ·paragraph·
      </div>
    </div>
  </div>
  <pre
    class="jsx-2255726786"
  >
    <span
      class="jsx-1782790912"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·3·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,1],"focus":[0,0,1]}}·
    </span>
    <span
      class="jsx-1782790912"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·2·
      · ·
      ·{"hasFocus":true,"selection":{"anchor":[0,0,0],"focus":[0,0,0]}}·
    </span>
    <span
      class="jsx-1782790912"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·1·
      · ·
      ·{"hasFocus":true,"selection":null}·
    </span>
    <span
      class="jsx-1782790912"
      title="{'tag':'div','props':{'className':'root'},'children':[{'tag':'div','props':{'style':{'fontSize':'24px'}},'children':[{'text':'heading'}]},{'tag':'div','props':{'style':{'fontSize':'16px'}},'children':[{'text':'paragraph'}]}]}"
    >
      ·0·
      · ·
      ·{"hasFocus":false,"selection":null}·
    </span>
  </pre>
</div>
`;
